core/templates/pages/exploration-player-page/services/question-player-engine.service.spec.ts

```diff
@@ -0,0 +1,1255 @@
+              html: '',
+              content_id: 'content'
+            },
+            interaction: {
+              customization_args: {},
+              answer_groups: [],
+              solution: null,
+              hints: [],
+              default_outcome: {
+                param_changes: [],
+                dest: 'Introduction',
+                feedback: {
+                  html: '',
+                  content_id: 'content'
+                },
+                labelled_as_correct: true,
+                refresher_exploration_id: 'exp',
+                missing_prerequisite_skill_id: null
+              },
+              confirmed_unclassified_answers: [],
+              id: null
+            }
+          }
+        }
+      },
+      version: 1,
+      can_edit: true,
+      preferred_audio_language_code: 'en',
+      preferred_language_codes: [],
+      auto_tts_enabled: true,
+      correctness_feedback_enabled: true,
+      record_playthrough_probability: 1
+    };
+  });
+
+  beforeEach(() => {
+    alertsService = TestBed.inject(AlertsService);
+    answerClassificationService = TestBed.inject(AnswerClassificationService);
+    contextService = TestBed.inject(ContextService);
+    expressionInterpolationService =
+      TestBed.inject(ExpressionInterpolationService);
+    questionObjectFactory = TestBed.inject(QuestionObjectFactory);
+    readOnlyExplorationBackendApiService = TestBed.inject(
+      ReadOnlyExplorationBackendApiService);
+    questionPlayerEngineService = TestBed.inject(QuestionPlayerEngineService);
+    urlService = TestBed.inject(UrlService);
+  });
+
+  it('should load questions when initialized', () => {
+    let successHandler = jasmine.createSpy('success');
+    let failHandler = jasmine.createSpy('fail');
+
+    let questionPlayerSpy =
+      spyOn(contextService, 'setQuestionPlayerIsOpen').and.returnValue(null);
+    let explorationIdSpy = spyOn(contextService, 'getExplorationId')
+      .and.returnValue('explorationId1');
+    let questionPlayerModeSpy =
+      spyOn(contextService, 'isInQuestionPlayerMode').and.returnValue(true);
+    let versionSpy =
+      spyOn(urlService, 'getExplorationVersionFromUrl').and.returnValue(1);
+
+    questionPlayerEngineService.init(
+      multipleQuestionBackendDict, successHandler, failHandler);
+
+    expect(questionPlayerSpy).toHaveBeenCalled();
+    expect(explorationIdSpy).toHaveBeenCalled();
+    expect(questionPlayerModeSpy).toHaveBeenCalled();
+    expect(versionSpy).toHaveBeenCalled();
+    expect(successHandler).toHaveBeenCalled();
+    expect(failHandler).not.toHaveBeenCalled();
+    expect(questionPlayerEngineService.getExplorationId())
+      .toBe('explorationId1');
+    expect(questionPlayerEngineService.questionPlayerMode).toBe(true);
+    expect(questionPlayerEngineService.getExplorationVersion()).toBe(1);
+    expect(questionPlayerEngineService.questions.length).toBe(3);
+  });
+
+  it('should update the exploration version if it' +
```

Comment:
Add a space after "it", otherwise you'll get "itis" in the terminal.

Reply:
updated here and every where else.